web/modules/custom/dpl_fees/src/Controller/DplFeesController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Drupal\dpl_fees\Controller;
+
+use Drupal\Core\Block\BlockManagerInterface;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Render\RendererInterface;
+use Drupal\dpl_react\DplReactConfigInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * Render fees list react app.
+ */
+class DplFeesController extends ControllerBase {
+
+  /**
+   * DplFeesController constructor.
+   *
+   * @param \Drupal\Core\Block\BlockManagerInterface $blockManager
+   *   Drupal block manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   Drupal renderer service.
+   * @param \Drupal\dpl_react\DplReactConfigInterface $feesSettings
+   *   Fees settings.
+   */
+  public function __construct(
+    private BlockManagerInterface $blockManager,
+    private RendererInterface $renderer,
+    private DplReactConfigInterface $feesSettings
+  ) {
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The Drupal service container.
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.block'),
+      $container->get('renderer'),
+      \Drupal::service('dpl_instant_loan.settings')
+    );
+  }
+
+  /**
+   * Display page with the fees list.
+   *
+   * @return mixed[]
+   *   Render array.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function list(): array {
+    $config = [];
+
+    /** @var \Drupal\dpl_fees\Plugin\Block\FeesListBlock $plugin_block */
+    $plugin_block = $this->blockManager->createInstance('dpl_fees_list_block', $config);
+
+    $access_result = $plugin_block->access($this->currentUser());
+    if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+      throw new AccessDeniedHttpException();
+    }
```

Comment:
Todo or common way of handling access...

Reply:
Added